test/fixtures/dialects/hive/select_interval.sql

```diff
@@ -0,0 +1,39 @@
+SELECT current_date + INTERVAL '2' DAY;
+
+SELECT current_date - INTERVAL '1' DAY AS yesterday;
+
+SELECT current_date + INTERVAL '12' HOUR;
+
+-- These examples are from:
+-- https://cwiki.apache.org/confluence/display/hive/languagemanual+types#LanguageManualTypes-Intervals
+SELECT INTERVAL '1' DAY;
+
+SELECT INTERVAL '1-2' YEAR TO MONTH;
+
+SELECT INTERVAL '1' YEAR + INTERVAL '2' MONTH;
+
+SELECT INTERVAL '1 2:3:4.000005' DAY;
+
+SELECT INTERVAL '1' DAY+
+       INTERVAL '2' HOUR +
+       INTERVAL '3' MINUTE +
+       INTERVAL '4' SECOND +
+       INTERVAL '5' NANO;
+
+SELECT INTERVAL 1 DAY;
+
+-- This interpretation may be too simplistic
+-- Original example was: INTERVAL (1+dt) DAY
+SELECT INTERVAL (1+2) DAY;
```

Comment:
This isn't being parsed by the IntervalExpressionSegment, it's being interpreted as a function:
![image](https://user-images.githubusercontent.com/80432516/146770331-4b95a137-90f3-4a28-8cd2-69fa920c8995.png)

(same thing happens if you do `INTERVAL (1+dt) DAY`

Reply:
To get both of those queries to parse we should have a bracketed expression in the grammar:
![image](https://user-images.githubusercontent.com/80432516/146772593-93e9d194-c164-4e27-a533-8f1193a34a51.png)

But the functions still seems to get matched first. @tunetheweb any idea on how we stop function from matching the `INTERVAL` keyword? (it's already a reserved keyword)

Reply:
I moved IntervalExpressionSegment to be in order before FunctionSegment in ansi_dialect. That seems to make parser try to match on INTERVAL before function and it seems like it may have done the trick. Please let me know..

Reply:
Ahh nice one, that's really good thinking! Looks good to me now 💪

Reply:
Would you be able to add in a unit test for the `INTERVAL (1+dt) DAY` example as well to see how that parses?

Reply:
That works too and I was switching to just have that. But I got some git trouble when trying to push because of https://github.com/sqlfluff/sqlfluff/pull/2144/commits/93e7b3c4c32932ef5468743800a4e18e7e0dea3d, oops :) But coming soon. There's also a fix for the NANO parsing.